test/extensions/filters/network/ext_authz/ext_authz_fuzz_test.cc

```diff
@@ -0,0 +1,123 @@
+    check_status = Filters::Common::ExtAuthz::CheckStatus::Error;
+    break;
+  }
+  case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusDenied: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::Denied;
+    break;
+  }
+  default: {
+    // Unhandled status
+    PANIC("A check status handle is missing");
+  }
+  }
+  return check_status;
+}
+
+DEFINE_PROTO_FUZZER(const envoy::extensions::filters::network::ext_authz::ExtAuthzTestCase& input) {
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  } catch (const ProtobufMessage::DeprecatedProtoFieldException& e) {
+    ENVOY_LOG_MISC(debug, "DeprecatedProtoFieldException: {}", e.what());
+    return;
+  }
+
+  Stats::TestUtil::TestStore stats_store;
+  Filters::Common::ExtAuthz::MockClient* client = new Filters::Common::ExtAuthz::MockClient();
+  envoy::extensions::filters::network::ext_authz::v3::ExtAuthz proto_config = input.config();
+
+  ConfigSharedPtr config = std::make_shared<Config>(proto_config, stats_store);
+  std::unique_ptr<Filter> filter =
+      std::make_unique<Filter>(config, Filters::Common::ExtAuthz::ClientPtr{client});
+
+  NiceMock<Network::MockReadFilterCallbacks> filter_callbacks;
+  filter->initializeReadFilterCallbacks(filter_callbacks);
+  Network::Address::InstanceConstSharedPtr addr =
+      std::make_shared<Network::Address::PipeInstance>("/test/test.sock");
+
+  ON_CALL(filter_callbacks.connection_, remoteAddress()).WillByDefault(ReturnRef(addr));
+  ON_CALL(filter_callbacks.connection_, localAddress()).WillByDefault(ReturnRef(addr));
+
+  for (const auto& action : input.actions()) {
+    switch (action.action_selector_case()) {
+    case envoy::extensions::filters::network::ext_authz::Action::kOnData: {
+      // Optional input field to set default authorization check result for the following "onData()"
+      if (action.on_data().has_result()) {
+        ON_CALL(*client, check(_, _, _, _))
+            .WillByDefault(WithArgs<0>(
+                Invoke([&](Filters::Common::ExtAuthz::RequestCallbacks& callbacks) -> void {
+                  callbacks.onComplete(makeAuthzResponse(
+                      resultCaseToCheckStatus(action.on_data().result().result_selector_case())));
+                })));
+      }
+      Buffer::OwnedImpl buffer(action.on_data().data());
+      filter->onData(buffer, action.on_data().end_stream());
+      break;
+    }
+    case envoy::extensions::filters::network::ext_authz::Action::kOnNewConnection: {
+      filter->onNewConnection();
+      break;
+    }
+    case envoy::extensions::filters::network::ext_authz::Action::kRemoteClose: {
+      filter_callbacks.connection_.raiseEvent(Network::ConnectionEvent::RemoteClose);
+      break;
+    }
+    case envoy::extensions::filters::network::ext_authz::Action::kLocalClose: {
```

Comment:
Is there a contract about this? If the we have a LocalClose and then try to call onData, what happens?

Reply:
> Is there a contract about this? If the we have a LocalClose and then try to call onData, what happens?

LocalClose and RemoteClose will have the same effect on the filter: if there is a call which is being processed, the filter will cancel it. And stats info will change `config_->stats().active_.dec();` No other effect on following onData().